ctms/acoustic_service.py

```diff
@@ -0,0 +1,279 @@
+import datetime
+import logging
+from typing import List
+
+from django.utils.encoding import force_bytes
+from lxml import etree
+from silverpop.api import Silverpop, SilverpopResponseException
+
+from ctms.schemas import ContactSchema, NewsletterSchema
+
+logger = logging.getLogger(__name__)
+
+
+class AcousticResources:
+    # TODO: externalize, maybe a DB table?
+    VALID_ACOUSTIC_MAIN_TABLE_FIELDS = {
+        "sfdc_id": 1,
+        "email": 1,
+        "amo_add_on_ids": 1,
+        "amo_display_name": 1,
+        "amo_email_opt_in": 1,
+        "amo_language": 1,
+        "amo_last_login": 1,
+        "amo_location": 1,
+        "amo_profile_url": 1,
+        "amo_user": 1,
+        "amo_user_id": 1,
+        "amo_username": 1,
+        "create_timestamp": 1,
+        "email_format": 1,
+        "email_id": 1,
+        "email_lang": 1,
+        "vpn_waitlist_geo": 1,
+        "vpn_waitlist_platform": 1,
+        "fxa_id": 1,
+        "fxa_account_deleted": 1,
+        "fxa_lang": 1,
+        "fxa_login_date": 1,
+        "fxa_first_service": 1,
+        "fxa_created_date": 1,
+        "fxa_primary_email": 1,
+        "double_opt_in": 1,
+        "has_opted_out_of_email": 1,
+        "mailing_country": 1,
+        "first_name": 1,
+        "last_name": 1,
+        "sumo_contributor": 1,
+        "sumo_user": 1,
+        "sumo_username": 1,
+        "basket_token": 1,
+        "unsubscribe_reason": 1,
+    }
+
+    MAIN_TABLE_SUBSCR_FLAGS = {
+        # maps the Basket/CTMS newsletter name to the Acoustic name
+        "common-voice": "sub_common_voice",
+        "mozilla-fellowship-awardee-alumni": "sub_mozilla_fellowship_awardee_alumni",
+        "firefox-accounts-journey": "sub_firefox_accounts_journey",
+        "firefox-news": "sub_firefox_news",
+        "mozilla-and-you": "sub_firefox_news",
+        "hubs": "sub_hubs",
+        "internet-health-report": "sub_internet_health_report",
+        "knowledge-is-power": "sub_knowledge_is_power",
+        "miti": "sub_miti",
+        "mixed-reality": "sub_mixed_reality",
+        "mozilla-festival": "sub_mozilla_festival",
+        "mozilla-foundation": "sub_mozilla_foundation",
+        "mozilla-technology": "sub_mozilla_technology",
+        "mozillians-nda": "sub_mozillians_nda",
+        "take-action-for-the-internet": "sub_take_action_for_the_internet",
+        "test-pilot": "sub_test_pilot",
+        "about-mozilla": "sub_about_mozilla",
+        "app-dev": "sub_apps_and_hacks",
+        "firefox-welcome": "firefox-welcome",  # TODO: Confirm approach, found newsletters through testing
+        "mozilla-welcome": "mozilla-welcome",  # TODO: Confirm
+        "firefox-os": "firefox-os",  # TODO: Confirm
+        "ambassadors": "ambassadors",  # TODO: Confirm
+        "maker-party": "maker-party",  # TODO: Confirm
+        "mozilla-learning-network": "mozilla-learning-network",  # TODO: Confirm
```

Comment:
If these were newsletters that were still active, then the Acoustic name would be "sub__something_".
If we need these in here for testing, then I'd change the Acoustic name to something like "FAKE_FOR_TESTING" so it's clear we're not actually using them. Or something?
I dunno, regardless, they shouldn't be passed on to Acoustic, since Acoustic doesn't know about them.

Reply:
Ok! I will do some testing modification. Just wanted to make sure this data-set of flags was up to date, was hoping to confirm/deny!